beacon-chain/sync/service_test.go

```diff
@@ -0,0 +1,49 @@
+package sync
+
+import (
+	"context"
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/internal"
+)
+
+func setupTestSyncService(t *testing.T) *Service {
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+
+	if err := db.InitializeState(); err != nil {
+		t.Fatalf("Failed to initialize state: %v", err)
+	}
+
+	cfg := &Config{
+		ChainService:  &mockChainService{},
+		P2P:           &mockP2P{},
+		BeaconDB:      db,
+		AttestService: &mockAttestService{},
+	}
+
+	return NewSyncService(context.Background(), cfg)
+}
+
+func TestStatusWithNetwork(t *testing.T) {
+	ss := setupTestSyncService(t)
+	synced, querierErr := ss.Querier.IsSynced()
+
+	if synced {
```

Comment:
The way the test is set up , synced will always be true since the `currentHeadSlot == 0` , so the test will never come to the path where synced is false